File: agent-driven-architecture-main/examples/typescript/test/gate/fixtures/violating/C16/src/spine/replay/replay.ts
// VIOLATION: G6 — the fold's ATTRIBUTED output opened outside the admission
// rule. Three spellings of ONE read, each of which would convert what the fold
// returned into something `keyedEffect` and `Sink.perform` accept, without the
// licence check `admit` performs (docs/DECISIONS.md:85).
//
// On the LIVE tree each of these is also a compile error, because `Attributed`
// holds both halves in `#`-private fields. That is the point of the pair: the
// language is the wall, and this rule is the TRIPWIRE that fires the moment a
// future author widens the field back into an ordinary public member.
export const dotted = (a) => a.emitted;
export const computed = (a) => a["emitted"];
export const destructured = (a) => {
  const { emitted } = a;
  return emitted;
};

File: agent-driven-architecture-main/examples/kotlin/src/test/fixtures/konsist/violating/C16/spine/replay/Replay.kt
// VIOLATION: G6 — the fold's ATTRIBUTED output opened outside the admission rule.
// The DOTTED property read, which on the live tree is also a COMPILE error because
// `Attributed` holds both halves privately. That is the point of the pair: the
// language is the wall, and this rule is the TRIPWIRE that fires the moment a
// future author widens the visibility back out.
//
// Two spellings a Kotlin author has that no text rule could safely see are absent
// on purpose and closed by the language instead: `with(a) { emitted }` needs a
// visible member, and `val (_, e) = a` needs `componentN()`, which a non-data
// class does not ship.

package adr.spine.replay

import adr.contract.Effect
import adr.spine.pure.Attributed

class Leak {

    fun dotted(attributed: Attributed): Effect = attributed.emitted

    fun spaced(attributed: Attributed): Effect = attributed. emitted
}
